docs/docusaurus/docs/core/trigger_actions_based_on_results/_examples/create_a_custom_action.py

```diff
@@ -0,0 +1,46 @@
+"""
+This is an example script for how to create a custom Action.
```

Comment:
**Note for technical reviewer:** this is my first time putting a new code sample under test. I did get some errors with it that I was able to resolve so I think the test is working. But, please let me know if I need to add anything else to make sure this is sufficiently tested.

Reply:
Hmm so usually these snippet files represent actual tests and I don't think we have any test logic here.

At the bottom of the file, could we instantiate an instance of this custom class and assert against its run method?

```
my_action = MyCustomAction(...)
result = my_action.run(...)
assert result == {"some": "info"}
``` 

You may need to create a checkpoint and that can be a little cumbersome so let me know if you want me to jump in and add these details on your behalf.

Reply:
Thank you - I would appreciate if you'd jump in and add the extra code to make this a sufficient test!

Reply:
I've updated the logic to show a concrete example of what a user might want to do but feel free to modify/critique as you see fit

Reply:
Thank you!

How does this custom calculated percentage of successful expectations across all validation results compare to the  `success_percent` info available through [`ExpectationSuiteValidationResult.statistics`](https://docs.greatexpectations.io/docs/reference/api/core/expectationsuitevalidationresult_class/)?

If these two things provide the same exact information in different ways, are there advantages to using a custom action instead of the built-in statistics?

Depending on how similar these two things are and what if any advantages there are to choosing one option over another in various scenarios, I think we should either acknowledge the similarities and tradeoffs or pick a different concrete example.

(this feedback is somewhat similar to the feedback you had for me about the email use case in the intro overlapping a bit with our built in email action.)

Reply:
connecting some dots: more discussion about this in slack at https://greatexpectationslabs.slack.com/archives/C05V0M18TEJ/p1734482967962709?thread_ts=1734360165.007409&cid=C05V0M18TEJ